### PR TITLE
Rename OpProtoAndCheckerMaker --> OpInfoMaker

### LINKED ISSUE #4450: Rename OpProtoAndCheckerMaker to OpInfoMaker
`OpProtoAndCheckerMaker` is too long and hard to understand. Change it to `OpInfoMaker`, because it actually fill fields in `OpInfo`

### PR DESCRIPTION
**Pull request by reyoung**
Fix #4450

### CODE REVIEW

File: paddle/operators/name_convention.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ To make the operator document itself more clear, we recommend operator names obe
 
 ### OpProtoMaker names
```
Comment (QiJune) [inline paddle/operators/name_convention.md]: OpProtoMaker names --> OpInfoMaker names

**Review by JiayiFeng — DISMISSED** [review]
A very careful work! Even changed `OpProtoMaker` in documents!

**Review by QiJune — DISMISSED** [review]
LGTM except one tiny typo error

File: paddle/operators/name_convention.md (hunk 1 above)
Comment (reyoung) [reply paddle/operators/name_convention.md]: Done.

**Review by QiJune — APPROVED** [review]
LGTM

**Comment by tonyyang-svail** [comment]
Discussed it with @reyoung , this PR will cause compilation failure on all WIP op porting. Better hold this PR for now.

**Pull request closed without merging**